clojure.c

```diff
@@ -0,0 +1,163 @@
+/**
+ *   Copyright (c) 2015, Miloslav Nenadál <nenadalm@gmail.com>
+ *
+ *   This source code is released for free distribution under the terms of the
+ *   GNU General Public License.
+ *
+ *   This module contains code for generating tags for the Clojure language.
+ */
+
+#include <string.h>
+
+#include "general.h"
+
+#include "parse.h"
+#include "read.h"
+#include "vstring.h"
+#include "entry.h"
+
+typedef enum {
+	K_FUNCTION,
+	K_NAMESPACE
+} clojureKind;
+
+static kindOption ClojureKinds[] = {
+	{TRUE, 'f', "function", "functions"},
+	{TRUE, 'n', "namespace", "namespaces"}
+};
+
+static vString *CurrentNamespace;
+
+static int isNamespace (const unsigned char *strp)
+{
+	return strncmp (++strp, "ns", 2) == 0 && isspace (strp[2]);
+}
+
+static int isFunction (const unsigned char *strp)
+{
+	return strncmp (++strp, "defn", 4) == 0 && isspace (strp[4]);
+}
+
+static int isQuote (const unsigned char *strp)
+{
+	return strncmp (++strp, "quote", 5) == 0 && isspace (strp[5]);
+}
+
+static void functionName (vString * const name, const unsigned char *dbp)
+{
+	const unsigned char *p;
+
+	if (*dbp == '\'')
+		dbp++;
+	else if (*dbp == '(' && isQuote (dbp))
+	{
+		dbp += 7;
+		while (isspace (*dbp))
+			dbp++;
+	}
```

Comment:
in clojure, you can define function ``hello`` like this ``(defn hello [])``
or create list with values ``defn, hello, []`` like this ``'(defn hello [])`` which is equivalent to ``(quote (defn hello []))``

so it's true quote check in the code is kind of useless

this PR only adds tags like high level function definitions with their namespaces
(can add nested functions by accident or can skip function if it's not written the way I never saw it like
```clojure
(
defn hello[])
```

So I could probably remove the quote thing for now.

```clojure
(ns 'somename)
```
will cast exception
``ClassCastException clojure.lang.PersistentList cannot be cast to clojure.lang.Symbol  clojure.lang.RT$1.invoke (RT.java:237)``